src/env-inl.h

```diff
@@ -325,6 +326,34 @@ inline void Environment::set_printed_error(bool value) {
   printed_error_ = value;
 }
 
+inline void Environment::print_sync_warn() const {
+  if (!warn_on_sync_)
+    return;
+
+  v8::HandleScope handle_scope(isolate_);
+  v8::Local<v8::StackTrace> stack = v8::StackTrace::CurrentStackTrace(
+      isolate_, 10, v8::StackTrace::kDetailed);
+
+  fprintf(stderr, "WARNING: Detected use of sync API\n");
+  for (int i = 0; i < stack->GetFrameCount() - 1; i++) {
```

Comment:
Could we use `Message::PrintCurrentStackTrace(isolate_, stderr)` so that v8 does the formatting? Since it's pretty complicated when you start to handle evals, constructors etc.